### DISCUSSION TITLE
RTG on amiberry-dev

### DISCUSSION THREAD

**Discussion by amyren1966** [BlitterStudio/amiberry#867] [Q&A] [answered]
I am currently using Amiberry 3.4 on my rpi4. I try to keep it updated, and I also regularly check the dev version to test the new betas.
I have set up a hdd with os3.9 and P96 installed to use with amiberry 3.4, and this works fine with RTG. (UAE gfx card)
But when I try to copy my config and hdd across to use with the dev versions, only PAL modes are available.
Is this a known issue, or is there something else I need to do to get RTG working with the beta?

**Comment by solskogen** [comment]
Take a look at the RTG page in the configuration GUI when running -dev. Make sure that you have the RTG board setup there.

**Reply by amyren1966** [reply]
Yes, I have that set up already. WIth UAE Zorro III, 4MB
RTG is also is listed in the hardware tab.

I noitce I wrote wrong version number for my current working setup. I use the latest v.4.1.6 where the RTG is working.
But if I try the dev version (currently v.4.2) using the exact same config, then I get only PAL screenmodes.

**Reply by midwan** [reply]
Please try after the latest commits today, there was an issue which only seemed to occur in some cases, causing the uaegfx card to fail initializing.

**Reply by amyren1966** [reply]
I am not able to compile the latest build now.
The output shows following messages: 

`src/audio.cpp: In function ‘void setdr(int, bool)’:
src/audio.cpp:1512:3: error: expected ‘}’ before ‘else’
   else {
   ^~~~
src/audio.cpp:1491:25: note: to match this ‘{’
  if (dmaen(DMA_MASTER)) {
                         ^
src/audio.cpp: At global scope:
src/audio.cpp:1516:2: error: expected unqualified-id before ‘else’
  else {
  ^~~~
src/audio.cpp:1522:1: error: expected declaration before ‘}’ token
 }
 ^
make: *** [<builtin>: src/audio.o] Error 1
make: *** Waiting for unfinished jobs....`

**Reply by midwan** [reply]
That was my bad, sorry. There were some errors left over from the merge, just fixed them.

**Reply by amyren1966** [reply]
Yes, it compiles now.
It did not improve the issue though.
I get this message in amigados, just before workbench is loaded:
Picasso96: Could not create graphics board  context for 'uaegfx'
please check your picasso96 installation

ps.
off topic obserervation: The default setup for the Competition Pro in gamecontrollersdb_user.txt have the a and b firebuttons swapped, so it is currently set for left-handed players.

**Reply by midwan** [reply]
That's strange, I only saw that error in Pimiga and nowhere else, and it was fixed there with the latest commits.
Have you tried with the latest from `master`? Have you tried doing a `make clean` first, to ensure that any old objects are cleared out, before compiling? Sometimes that helps.

Regarding the Comp. Pro joystick setup, it seems to be working correctly in my tests as well: the Left fire button is Fire1, when I tested it in TinyBobble.

**Comment by amyren1966** [comment]
My normal Amiberry from master is 4.1.6 (06-dec.21), and on the master RTG is working, it is the dev that doesnt.
I did try to do a make clean before re-compiling it, but still no rtg.

**Reply by midwan** [reply]
Please try again after the latest commits in `dev` today? I synced the p96 file with `master`, so even though I cannot recreated, it should at least behave the same way.

**Comment by amyren1966** [comment, marked as answer]
Thanks, well done. 
RTG work as it should now, both with 4.2 and on the dev 4.3beta
